Modules/timemodule.c

```diff
@@ -2213,6 +2214,19 @@ static int
 pysleep(PyTime_t timeout)
 {
     assert(timeout >= 0);
+    assert(!PyErr_Occurred());
+#ifndef MS_WINDOWS
+    if (timeout == 0) { // gh-125997
```

Comment:
This test also works on Windows. `_PyTime_As100Nanoseconds(timeout, _PyTime_ROUND_CEILING)` can only return zero is timeout is zero.

Reply:
Actually, it would but `_PyTime_As100Nanoseconds` is only defined on Windows:

```c
#ifdef MS_WINDOWS
PyTime_t
_PyTime_As100Nanoseconds(PyTime_t ns, _PyTime_round_t round)
{
    return pytime_divide(ns, NS_TO_100NS, round);
}
#endif
```

Should I undef it?